internal/connector/internal/config/connectors.go

```diff
@@ -51,20 +58,82 @@ func NewConnectorsConfig() *ConnectorsConfig {
 
 func (c *ConnectorsConfig) AddFlags(fs *pflag.FlagSet) {
 	fs.StringArrayVar(&c.ConnectorCatalogDirs, "connector-catalog", c.ConnectorCatalogDirs, "Directory containing connector catalog entries")
+	fs.StringArrayVar(&c.ConnectorMetadataDirs, "connector-metadata", c.ConnectorMetadataDirs, "Directory containing connector metadata configuration files")
 	fs.DurationVar(&c.ConnectorEvalDuration, "connector-eval-duration", c.ConnectorEvalDuration, "Connector eval duration in golang duration format")
 	fs.StringArrayVar(&c.ConnectorEvalOrganizations, "connector-eval-organizations", c.ConnectorEvalOrganizations, "Connector eval organization IDs")
 	fs.BoolVar(&c.ConnectorNamespaceLifecycleAPI, "connector-namespace-lifecycle-api", c.ConnectorNamespaceLifecycleAPI, "Enable APIs to create, update, delete non-eval Namespaces")
 	fs.BoolVar(&c.ConnectorEnableUnassignedConnectors, "connector-enable-unassigned-connectors", c.ConnectorEnableUnassignedConnectors, "Enable support for 'unassigned' state for Connectors")
 }
 
 func (c *ConnectorsConfig) ReadFiles() error {
-	typesLoaded := map[string]string{}
-	var values []ConnectorCatalogEntry
+	// read metadata first to merge with catalog next
+	connectorMetadata, err := c.readConnectorMetadata()
+	if err != nil {
+		return err
+	}
+
+	c.CatalogEntries, err = c.readConnectorCatalog(connectorMetadata)
```

Comment:
It seems to that this assignment is the only reason why ConnectorsConfig is implementing this function by reference instead of value, by the look of it a value implementation will be enough and yield the same results if only we could get rid of this assignment.

Maybe we can return CatalogEntries instead of assigning it?

Reply:
I am not sure I understand what you saying. But I'll optimize it to directly assign `values` to `c.CatalogEntries` in the function. I think that's what you are suggesting.

Reply:
No really. My worry is that I see `(c *ConnectorsConfig)` being the receiver instead of `(c ConnectorsConfig)`. I don't see a reason why a pointer should be the receiver instead of a variable and the only change required to change the receiver from reference to value would be to get rid of the line `c.CatalogEntries, err = c.readConnectorCatalog(connectorMetadata)` and returning the value instead.

Reply:
Fixed.